lightningd/plugin.c

```diff
@@ -503,49 +503,30 @@ static const char *plugin_log_handle(struct plugin *plugin,
 
 	call_notifier = (level == LOG_BROKEN || level == LOG_UNUSUAL)? true : false;
 
-	/* Unescape log message. */
-	tal_t *ctx = tal(NULL, char);
-	const char *log_escaped = plugin->buffer + msgtok->start;
-	const size_t log_escaped_len = msgtok->end - msgtok->start;
-	struct json_escape *esc = json_escape_string_(ctx, log_escaped, log_escaped_len);
-	const char *log_msg = json_escape_unescape(ctx, esc);
-
-	/* Find last line. */
-	const char *log_msg2 = log_msg;
-	const char *last_msg;
-	while (true) {
-		const char *msg_end = strchr(log_msg2, '\n');
-		if (!msg_end) {
-			break;
-		}
-		int msg_len = msg_end - log_msg2;
-		if (msg_len > 0) {
-			last_msg = log_msg2;
-		}
-		log_msg2 = msg_end + 1;
-	}
+	/* Only bother unescaping and splitting if it has \ */
+	if (memchr(plugin->buffer + msgtok->start, '\\', msgtok->end - msgtok->start)) {
+		const char *log_escaped = plugin->buffer + msgtok->start;
+		const size_t log_escaped_len = msgtok->end - msgtok->start;
+		struct json_escape *esc = json_escape_string_(tmpctx, log_escaped, log_escaped_len);
+		const char *log_msg = json_escape_unescape(tmpctx, esc);
```

Comment:
I'm not sure how this fits into the bigger picture, but is unescaping everything here the intention?  I'm all for fewer elder backslashes, but it also results in some tests breaking due to double escaped `"` becoming single escaped for instance.